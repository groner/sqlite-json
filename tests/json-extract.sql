-- Should extract [1,2,3]
SELECT JSON_EXTRACT('{"foo": [1, 2, 3]}', 'foo') AS 'foo';
-- Should extract 2
SELECT JSON_EXTRACT('{"foo": [1, 2, 3]}', 'foo', 1) AS 'foo, 1';
-- Should extract 3
SELECT JSON_EXTRACT('{"foo": [1, 2, 3]}', 'foo', -1) AS 'foo, -1';
-- Should be SQL NULL
SELECT JSON_EXTRACT('{"foo": [1, 2, 3]}', 'bar') AS 'bar';
SELECT JSON_EXTRACT('{"foo": [1, 2, 3]}', 'bar') IS NULL AS 'bar IS NULL';
-- Should be null
SELECT JSON_EXTRACT('{"foo": [1, 2, 3], "bar": null}', 'bar') AS 'bar';
SELECT JSON_EXTRACT('{"foo": [1, 2, 3], "bar": null}', 'bar') IS NOT NULL AS 'bar IS NOT NULL';
