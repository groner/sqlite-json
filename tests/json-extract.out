-- Should extract [1,2,3]
SELECT JSON_EXTRACT('{"foo": [1, 2, 3]}', 'foo') AS 'foo';
foo       
----------
[1,2,3]   
-- Should extract 2
SELECT JSON_EXTRACT('{"foo": [1, 2, 3]}', 'foo', 1) AS 'foo, 1';
foo, 1    
----------
2         
-- Should extract 3
SELECT JSON_EXTRACT('{"foo": [1, 2, 3]}', 'foo', -1) AS 'foo, -1';
foo, -1   
----------
3         
-- Should be SQL NULL
SELECT JSON_EXTRACT('{"foo": [1, 2, 3]}', 'bar') AS 'bar';
bar       
----------
NULL      
SELECT JSON_EXTRACT('{"foo": [1, 2, 3]}', 'bar') IS NULL AS 'bar IS NULL';
bar IS NULL
-----------
1          
-- Should be null
SELECT JSON_EXTRACT('{"foo": [1, 2, 3], "bar": null}', 'bar') AS 'bar';
bar       
----------
null      
SELECT JSON_EXTRACT('{"foo": [1, 2, 3], "bar": null}', 'bar') IS NOT NULL AS 'bar IS NOT NULL';
bar IS NOT NULL
---------------
1              
